src/Cardiompp.Application/Services/DoctorService.cs

```diff
@@ -42,19 +45,39 @@ GetDoctorByEmailAndPasswordRequest loginRequest
             return new DoctorResponse<GetDoctorByEmailAndPasswordResponse>(doctor?.ToGetByEmailAndPasswordDataContact());
         }
 
-        public async Task<bool> UpdatePassword
+        public async Task<BaseResponse> UpdatePassword
         (
             UpdatePasswordRequest updatePasswordRequest
         ) 
         {
             var passwordMd5 = HashService.GetMd5Hash(updatePasswordRequest.Password);
             var newPasswordMd5 = HashService.GetMd5Hash(updatePasswordRequest.NewPassword);
             var doctor = await _unitOfWork.DoctorRepository.GetByEmailAndPassword(updatePasswordRequest.Email, passwordMd5);
+            var baseResponse = new BaseResponse();
+
+            if (doctor == null)
+            {
+                var passwordError = new ErrorResponse()
+                {
+                    Code = 400,
```

Comment:
Você não precisa trabalhar com código de erros na sua API, mas se quiser, cada erro da API é um código diferente. Se você começou em 400, o próximo vai ter que ser 401 e etc. Não sei se a sua ideia de colocar 400 era por causa do `BadRequest`, mas você não precisa seguir esse padrão do http aqui, até pq se você tiver mais de 100 erros você já vai pular pra casa do 500. Talvez seja melhor começar em 100, ou sei lá.

Se você for ter código de erro para cada validação, é bom centralizar isso em uma estrutura como um `Enum` ou `struct`, pra usar tipo `ValidationErrorCode.EmailOrPasswordInvalid` ou algo do tipo.

Reply:
Entendi, acho que faz sentido.
![image](https://user-images.githubusercontent.com/12301754/82722946-4fc1f380-9ca1-11ea-81a3-8229dad532ff.png)
![image](https://user-images.githubusercontent.com/12301754/82722954-57819800-9ca1-11ea-856f-4661a7841eef.png)
Mais pra frente pretendo implementar uma classe de resources para armazenar as mensagens de erro.